torch_xla/csrc/tensor_util.cpp

```diff
@@ -893,6 +926,18 @@ std::vector<torch::lazy::BackendDataPtr> CreateTensorsData(
     // populate_fn to be called.
     populate_mwait->Wait();
     return async->async_datas;
+  } else if (ShardingUtil::UseVirtualDevice()) {
```

Comment:
Let's make move this `if` block to the top, before we check for `if(transfer_async)`. This can be an independent terminal condition, no `if-else`.